base/Users.py

```diff
@@ -2,50 +2,42 @@
 from configparser import ConfigParser as Config
 from configparser import ExtendedInterpolation
 
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ObjectDoesNotExist
 from evaluation_system.misc import config
 from evaluation_system.model.db import UserDB
 from evaluation_system.model.user import User
 
 from base.exceptions import UserNotFoundError
-from django_evaluation.ldaptools import get_ldap_object
 
 
-class LdapUser(User):
-    """The freva evaluation system works with user accounts which are known to the underlying
-    operating system. This is not necessarily true for the web interface as it can be started
-    inside a Docker container without any access to a user directory. To bridge this gap
-    the web interface has its own version of the `User` class which directly connects to the
-    configured LDAP instead of asking the operating system for user information.
-    """
+class OpenIdUser(User):
+    """Get user information from JSON Web Token."""
 
-    def __init__(self, username):
-        ldap_object = get_ldap_object()
-        user_info = ldap_object.get_user_info(username)
-        if not user_info:
-            raise UserNotFoundError()
+    def __init__(self, username: str):
+        try:
+            _user_model = get_user_model()
+            _user = _user_model.objects.get(username=username)
+        except ObjectDoesNotExist as error:
+            raise UserNotFoundError() from error
         self._dir_type = config.get(config.DIRECTORY_STRUCTURE_TYPE)
 
         self._username = username
-        self._uid = user_info[0]
-        self._email = user_info[3]
-        self._home_directory = user_info[4]
+        self._uid = "web"
```

Comment:
as we are defining constant values here and in `_home_directory`, and on the other hand we are not supposed to use ldap anymore, i think in anohter branch we need to refactor stuff a bit to get rid of these two constant vars and `getUserId` and `getUserHome`. But i don't see it mandatory in this PR. @Karinon ur idea?

Reply:
This user model is defined in [freva core](https://github.com/FREVA-CLINT/freva/blob/04807a98c6acc2147c3e41ee5c5788dab4b12288/src/evaluation_system/model/user.py#L52) which entirely relies on getting the information from the operating system - I am not sure where, most likely for loading the plugins, a freva user object needs to be created. But we can't simply use the freva user class because the uid/username doesn't exsit in the docker container. So what @Karinon was doing it creating another user class that mimicks the behaviour of the freva user class but get's its information from ldap. And I simply changed that towards getting the user information from the openid-connect service.

Reply:
true, i just went through the code and found, it's super nested and on the other hand shaking this func a bit, can make a earthquake in whole auth, so, i was doubt and ask for changing in another PR and appropriate time ...